mosviz/controls/slit_selection_ui.py

```diff
@@ -0,0 +1,215 @@
+    will be lost.
+    """
+    def __init__(self, mosviz_viewer, parent=None):
+        super(SlitSelectionUI, self).__init__(parent=parent)
+
+        self.mosviz_viewer = mosviz_viewer
+        self._slit_dict = {}
+
+        self._mosviz_table_option_text = 'Slit from MOSViz Table'
+
+        self._init_ui()
+
+    def _init_ui(self):
+        self.slit_type_label = QLabel('Slit Type')
+        self.slit_type_combo = QComboBox()
+        self.slit_type_combo.currentIndexChanged.connect(self.update_info)
+
+        hbl1 = QHBoxLayout()
+        hbl1.addWidget(self.slit_type_label)
+        hbl1.addWidget(self.slit_type_combo)
+
+        self.slit_width_label = QLabel('Slit Width')
+        self.slit_width_input = QLineEdit()
+        self.slit_width_combo = QComboBox()
+        self.slit_width_units = QLabel('arcsec')
+
+        hbl2 = QHBoxLayout()
+        hbl2.addWidget(self.slit_width_label)
+        hbl2.addWidget(self.slit_width_input)
+        hbl2.addWidget(self.slit_width_combo)
+        hbl2.addWidget(self.slit_width_units)
+
+        self.slit_length_label = QLabel('Slit Length')
+        self.slit_length_input = QLineEdit()
+        self.slit_length_combo = QComboBox()
+        self.slit_length_units = QLabel('arcsec')
+
+        hbl3 = QHBoxLayout()
+        hbl3.addWidget(self.slit_length_label)
+        hbl3.addWidget(self.slit_length_input)
+        hbl2.addWidget(self.slit_length_combo)
+        hbl3.addWidget(self.slit_length_units)
+
+        self.okButton = QPushButton('Apply')
+        self.okButton.clicked.connect(self.apply)
+        self.okButton.setDefault(True)
+
+        self.cancelButton = QPushButton('Cancel')
+        self.cancelButton.clicked.connect(self.cancel)
+
+        hbl4 = QHBoxLayout()
+        hbl4.addWidget(self.cancelButton)
+        hbl4.addWidget(self.okButton)
+
+        vbl = QVBoxLayout()
+        vbl.addLayout(hbl1)
+        vbl.addLayout(hbl2)
+        vbl.addLayout(hbl3)
+        vbl.addLayout(hbl4)
+        self.setLayout(vbl)
+        self.vbl = vbl
+
+        self._load_selections()
+        self._populate_combo()
+
+        self.show()
+
+    def _load_selections(self):
+        """Load preconfigured slit shapes from yaml file"""
+        file_path = os.path.join(os.path.dirname(__file__), 'saved_slits.yaml')
+        with open(file_path) as f:
+            self.slit_dict = yaml.load(f)
+
+    def _populate_combo(self, default_index=0):
+        """Populate combo box with slit types"""
+        name_list = [self._mosviz_table_option_text] + \
+                    [self.slit_dict[s]['name'] for s in sorted(self.slit_dict)] + \
+                    ['Custom']
+
+        key_list = ['default'] + [s for s in sorted(self.slit_dict)] + ['custom']
+
+        combo_input = [(name, key) for name, key in zip(name_list, key_list)]
+        update_combobox(self.slit_type_combo, combo_input, default_index=default_index)
+
+    def get_width(self):
```

Comment:
Unless this is overloading a method, you should convert this to a property `width`.  The Python convention is to aim for getters that are properties rather than using `get_<variable>` type of thing.